### PR TITLE
jason first version without full unit test

### PR DESCRIPTION
**Pull request by jasonyang66**
@rysiekblah

### CODE REVIEW

File: src/main/java/com/epam/processor/DataProcessor.java [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
 
 import com.epam.data.RoadAccident;
+import com.epam.data.RoadAccidentBuilder;
+import com.google.common.collect.ArrayListMultimap;
 import com.google.common.collect.Multimap;
 
-import java.util.Collection;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
```
Comment (rysiekblah) [inline src/main/java/com/epam/processor/DataProcessor.java]: I suggest not include all utils

File: src/main/java/com/epam/processor/DataProcessor.java [hunk 2]
```diff
@@ -27,7 +29,8 @@ public DataProcessor(List<RoadAccident> roadAccidentList){
      * @return
      */
     public RoadAccident getAccidentByIndex7(String index){
-        return null;
+        RoadAccidentBuilder builder = new RoadAccidentBuilder(index);
```
Comment (rysiekblah) [inline src/main/java/com/epam/processor/DataProcessor.java]: this is wrong, you should query existing list in this class: roadAccidentList

File: src/main/java/com/epam/processor/DataProcessor.java [hunk 3]
```diff
@@ -40,6 +43,18 @@ public RoadAccident getAccidentByIndex7(String index){
      * @return
      */
     public Collection<RoadAccident> getAccidentsByLocation7(float minLongitude, float maxLongitude, float minLatitude, float maxLatitude){
+        Collection<RoadAccident> roadAccidentList = new ArrayList<RoadAccident>();
+        if(this.roadAccidentList.size()>0) {
```
Comment (rysiekblah) [inline src/main/java/com/epam/processor/DataProcessor.java]: you dont have to check this, if size is 0, there will be 0 loop iteration

File: src/main/java/com/epam/processor/DataProcessor.java [hunk 4]
```diff
@@ -40,6 +43,18 @@ public RoadAccident getAccidentByIndex7(String index){
...
+            for(RoadAccident roadAccident:roadAccidentList){
+                if(roadAccident.getLongitude() < minLongitude || roadAccident.getLatitude() < minLatitude
+                        || roadAccident.getLongitude() > maxLongitude || roadAccident.getLatitude() > maxLatitude) {
```
Comment (rysiekblah) [inline src/main/java/com/epam/processor/DataProcessor.java]: if you change this condition you can avoid using "else"

File: src/main/java/com/epam/processor/DataProcessor.java [hunk 5]
```diff
@@ -51,15 +66,78 @@ public RoadAccident getAccidentByIndex7(String index){
      * @return
      */
     public Map<String, Long> getCountByRoadSurfaceCondition7(){
+        Map<String,Long> roadSurfaceMap = new TreeMap<String,Long>();
+        Long firstElement = 1L;
+        if(this.roadAccidentList.size()>0) {
```
Comment (rysiekblah) [inline src/main/java/com/epam/processor/DataProcessor.java]: same here, no need to check list size

**Comment by jasonyang66** [comment]
Ok, will modify it, anyway the command you just filled in to set the github, could you please help to input here again, thanks!

Best regards,
Jason



来自 魅蓝 note 3



-------- 原始邮件 --------
发件人：Tomasz <notifications@github.com>
时间：周四 12月1日 13:40
收件人：java-mentoring-program-2/RoadAccidentsMgmt <RoadAccidentsMgmt@noreply.github.com>
抄送：Jason Yang <Jason_Yang@epam.com>,Author <author@noreply.github.com>
主题：Re: [java-mentoring-program-2/RoadAccidentsMgmt] jason first version without full unit test (#16)


@rysiekblah commented on this pull request.

________________________________

In src/main/java/com/epam/processor/DataProcessor.java<https://github.com/java-mentoring-program-2/RoadAccidentsMgmt/pull/16#pullrequestreview-10904546>:

> @@ -27,7 +29,8 @@ public DataProcessor(List<RoadAccident> roadAccidentList){
      * @return
      */
     public RoadAccident getAccidentByIndex7(String index){
-        return null;
+        RoadAccidentBuilder builder = new RoadAccidentBuilder(index);


this is wrong, you should query existing list in this class: roadAccidentList

File: src/main/java/com/epam/processor/DataProcessor.java [hunk 6]
```diff
@@ -51,15 +66,78 @@ public RoadAccident getAccidentByIndex7(String index){
      * @return
      */
     public Map<String, Long> getCountByRoadSurfaceCondition7(){
+        Map<String,Long> roadSurfaceMap = new TreeMap<String,Long>();
+        Long firstElement = 1L;
```
Comment (rysiekblah) [inline src/main/java/com/epam/processor/DataProcessor.java]: do you really need this "firstElement" var?

File: src/main/java/com/epam/processor/DataProcessor.java [hunk 7]
```diff
@@ -51,15 +66,78 @@ public RoadAccident getAccidentByIndex7(String index){
...
+
+        /**
+         * find the weather conditions which caused the top 3 number of incidents
+         * as example if there were 10 accidence in rain, 5 in snow, 6 in sunny and 1 in foggy, then your result list should contain {rain, sunny, snow} - top three in decreasing order
+         * @return
+         */
     public List<String> getTopThreeWeatherCondition7(){
+
+
+        Map<String,Long> roadSurfaceMap = new TreeMap<String,Long>();
```
Comment (rysiekblah) [inline src/main/java/com/epam/processor/DataProcessor.java]: I think you can use SortedMap

**Pull request closed without merging**